### PR TITLE
[tools][Android] Add android-update-native-dependencies command

### PR DESCRIPTION
**Pull request by bbarthec**
# Why

Our Android native dependencies are quite out-of-date.
To keep them up to date we have to open every single `gradle.build` file, locate dependency and bump it's version (Android Studio also has some mechanism for updating dependencies, but I've failed to see all the dependencies in it 😞).

I've concluded that (following Brent's suggestion) that maybe we'll benefit from some tooling for this process.
There is a tool out there: https://github.com/kevcodez/gradle-upgrade-interactive, but it is lacking in many areas (eg. no option to change the suggested version of the dependency).
This tool bases on https://github.com/ben-manes/gradle-versions-plugin that seems very fancy and I followed along with it by creating a separate `expotools command` that tries to gather all the dependencies in all android sub-projects we have in `expo/expo` monorepo.

Additionally, what I think is most critical, searching every time for libraries changelogs is very dull process, so I've created `dependenciesChangelog.json` file that contains encountered Android native dependencies. This file is leveraged by the proposed command as well.

# How

I've created an expotools command `android-update-native-dependencies` that does the followings:
- launches `gradle updateDependency` task provided by https://github.com/ben-manes/gradle-versions-plugin for Expo Go project and `bare-expo` project (the latter includes some more `packages/*` only available in bare workflow). This task produces some `json` file describing detected dependencies is each android project.
- caches the results basing on daily timestamps and command params (I've checked that changing the `revision` gradle task params brings no visible change to the results produces by this task, bu who knows for sure 🤷)
- reads these projects' dependencies descriptions and presents them for some selection
- enables manual version input for each library (presents links with each dependency's changelog)
- modifies `gradle` files according to the final decision
- adds changelogs entries in each package describing the native dependency update with the link to the dependency's changelog.

# Test Plan

Just run `et android-update-native-dependencies -h` and follow along.

# Showcase

command

# TODOs

- [ ] Create a similar tool for iOS native dependencies.
- [ ] Use this tool to bump native dependencies with some mindful strategy 🧘. (PRs incoming 🛬)

### CODE REVIEW

File: tools/src/android-update-native-dependencies/index.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    }
+
+    if (unresolvedExceededDependencies.length > 0) {
+      logger.log(chalk.yellow('  ◼︎ unresolved/exceeded dependencies:'));
+      unresolvedExceededDependencies.forEach(printDependency);
+    }
+  }
+}
+
+export interface Options {
```
Comment (tsapeta) [inline tools/src/android-update-native-dependencies/index.ts:46]: Why not a type? I think it's not going to be implemented by any class 🤔

File: tools/src/android-update-native-dependencies/dependenciesChangelogs.json [hunk 2]
```diff
@@ -0,0 +1,116 @@
+{
```
Comment (tsapeta) [inline tools/src/android-update-native-dependencies/dependenciesChangelogs.json:1]: We will need to find a better place for such jsons 🤔 We also have `changelogVersions.json` in the root folder (which is updated by `et merge-changelogs`) — might be good to keep them close — maybe `tools/data`?. What do you think?

**Review by tsapeta — CHANGES_REQUESTED** [review]
Good job Bartek! 👏 💪 
Left a few comments, but I really like what you've done! 😍

**Review by barthap — COMMENTED** [review]
Not sure if you requested my review on purpose or accidentally, but anyway I took a quick look and left some comments 😅

**Pull request closed without merging**